tools/integration_tests/gzip/helpers/helpers.go

```diff
@@ -0,0 +1,177 @@
+// If gzipCompress is true, output file is a gzip-compressed file.
+// contentSize is the size of the uncompressed content. In case gzipCompress is true, the actual output file size will be
+// different from contentSize (typically gzip-compressed file size < contentSize).
+// Caller is responsible for deleting the created file when done using it.
+// Failure cases:
+// 1. contentSize <= 0
+// 2. os.CreateTemp() returned error or nil handle
+// 3. gzip.NewWriter() returned nil handle
+// 4. Failed to write the content to the created temp file
+func CreateLocalTempFile(contentSize int, gzipCompress bool) (string, error) {
+	// fail if contentSize <= 0
+	if contentSize <= 0 {
+		return "", fmt.Errorf("unsupported fileSize: %d", contentSize)
+	}
+
+	// Create text-content of given size.
+	// strings.builder is used as opposed to string appends
+	// as this is much more efficient when multiple concatenations
+	// are required.
+	var contentBuilder strings.Builder
+	const tempStr = TempFileStrLine + "\n"
+
+	for ; contentSize >= len(tempStr); contentSize -= len(tempStr) {
+		contentBuilder.WriteString(tempStr)
+	}
+
+	if contentSize > 0 {
+		contentBuilder.WriteString(tempStr[0:contentSize])
+	}
+
+	// reset contentSize
+	contentSize = contentBuilder.Len()
+
+	// create appropriate name template for temp file
+	filenameTemplate := "testfile-*.txt"
+	if gzipCompress {
+		filenameTemplate += ".gz"
+	}
+
+	// create a temp file
+	f, err := os.CreateTemp(TmpDirectory, filenameTemplate)
+	if err != nil {
+		return "", err
+	} else if f == nil {
+		return "", fmt.Errorf("nil file handle returned from os.CreateTemp")
+	}
+	defer operations.CloseFile(f)
+	filepath := f.Name()
+
+	content := contentBuilder.String()
+
+	if gzipCompress {
+		w := gzip.NewWriter(f)
+		if w == nil {
+			return "", fmt.Errorf("failed to open a gzip writer handle")
+		}
+		defer func() {
+			err := w.Close()
+			if err != nil {
+				fmt.Printf("Failed to close file %s: %v", filepath, err)
+			}
+		}()
+
+		// write the content created above as gzip
+		n, err := w.Write([]byte(content))
+		if err != nil {
+			return "", err
+		} else if n != contentSize {
+			return "", fmt.Errorf("failed to write to gzip file %s. Content-size: %d bytes, wrote = %d bytes", filepath, contentSize, n)
+		}
+	} else {
+		// write the content created above as text
+		n, err := f.WriteString(content)
+		if err != nil {
+			return "", err
+		} else if n != contentSize {
+			return "", fmt.Errorf("failed to write to text file %s. Content-size: %d bytes, wrote = %d bytes", filepath, contentSize, n)
+		}
+	}
+
+	return filepath, nil
+}
+
+// Downloads given gzipped GCS object (with path without 'gs://') to local disk.
+// Fails if the object doesn't exist or permission to read object is not
+// available.
+// Uses go storage client library to download object.
+func DownloadGzipGcsObjectAsCompressed(bucketName, objPathInBucket string) (string, error) {
```

Comment:
is there any reason why we are not using gcloud/gsutil here to download gzip file?

Reply:
Yes, that was my first instinct as well, but gcloud/gsutil don't support reading back a gzip object as compressed. I tried with all arguments that I knew of, but none works, and they always return the decompressed content on `gsutil cp` or `gcloud storage cp`. So, I had to fallback to go-storage-library, same as I have done in the actual implementation.

Reply:
Got it, I think it's worth mentioning that as comment.